GoogleMapsApi.Test/IntegrationTests/ElevationTests.cs

```diff
@@ -13,25 +13,12 @@ public void Elevation_ReturnsCorrectElevation()
         {
             var request = new ElevationRequest { Locations = new[] { new Location(40.7141289, -73.9614074) } };
 
-            var result = GoogleMaps.Elevation.Query(request);
+            var result = GoogleMaps.Elevation.Query(request).Result;
 
             if (result.Status == Entities.Elevation.Response.Status.OVER_QUERY_LIMIT)
                 Assert.Inconclusive("Cannot run test since you have exceeded your Google API query limit.");
             Assert.AreEqual(Entities.Elevation.Response.Status.OK, result.Status);
             Assert.AreEqual(14.78, result.Results.First().Elevation, 1.0);
         }
-
-        [Test]
-        public void ElevationAsync_ReturnsCorrectElevation()
```

Comment:
Since the non async one is async now, it appeared that that test added no value over the main one since that was upgraded to do the same test effectively.

I noticed as I was going along (I didn't intend to touch the async stuff) that MS in their libs tend to have async only and don't have sync methods anymore. Example HttpClient.cs . Since I was using that lib to replace WebClient.cs it made sense to align them. I think one mistake I made here is perhaps that MS use the pattern: QueryAsync and not Query. Whereas I used Query but not QueryAsync. I can probably rename this weekend.

The power of Async being backed to tasks now is that the user has significant control over the tasks that were offered by your sync methods previously. Tasks are more abstract and allow the user a more familar way of dealing with syncronicity. Perhaps new factories could be introduced that help add task syncroncity extensions etc.